### DISCUSSION TITLE
Secondary Engine

### DISCUSSION THREAD

**Discussion by lefred** [stoneatom/stonedb#921] [Ideas]
Have you considered using stonedb as a secondary engine instead (in MySQL 8.0)?

**Comment by Nliver** [comment]
Oh! @lefred ，I'm glad you're following our work, and I want to tell you this:
yeah~ Of course, one clear time point is that **we expect MySQL 8.0 to be officially supported around June 2023**~
You are welcome to leave your comments on StoneDB, Thanks.

**Comment by RingsC** [comment]
Sorry for late replying your question.  Yes, definitely, We found the secondary engine is a very attarctive new feature in MySQL 8.0. By analyzing the MySQL Heatwave, we assume that The Heatwave `RAPID` engine is also a secondary engine. And also it employes an in-memory column memory engine to enable HTAP feature in an integrated system.  Therefore, we also agree with this idea, and We will implement the secondary engine into StoneDB version 2.0, and an in-memory column store engine also will be introduced. Now, the Tianmu engine is a on-disk column store engine.  And, RFCs onf StoneDB version 2.0 is availbe now. 

Overview of StoneDB version 2.0 is listed here for your information.  https://github.com/stoneatom/stonedb/issues/436
  
Thanks for your watching us.  
Best Regards.
